### PR TITLE
Add new test case, Hibernate VM size change

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
Change the VM size from D8s_v3 to D4s_v3.
Expected the VM restarting due to the different memory size.
Verify the expected log messages.

**TEST RESULT**

[LISAv2 Test Results Summary]
Test Run On           : 06/22/2020 06:50:33
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Initial Kernel Version: 4.15.0-1089-azure
Final Kernel Version  : 5.8.0-rc1-next-20200622
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:37

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                POWER-VMRESIZE-HIBERNATE                                                          PASS                35.41 

[LISAv2 Test Results Summary]
Test Run On           : 06/22/2020 06:50:32
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Initial Kernel Version: 5.3.0-1028-azure
Final Kernel Version  : 5.8.0-rc1-next-20200622
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:40

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                POWER-VMRESIZE-HIBERNATE                                                          PASS                38.07

### CODE REVIEW

File: Testscripts/Windows/VMRESIZE-HB.ps1 [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+			$state = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password -command "date > /dev/null; echo $?"
+			if ($state -eq 0) {
+				Write-LogInfo "VM is not accessible remotely"
+				break
+			} else {
+				Write-LogInfo "VM is still resuming!"
+			}
+		}
+
+		if ($vmCount -le 0){
```
Comment (LiliDeng) [inline Testscripts/Windows/VMRESIZE-HB.ps1]: did you operate this in above loop?

File: Testscripts/Windows/VMRESIZE-HB.ps1 [hunk 2]
```diff
@@ -0,0 +1,255 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+<#
+.Synopsis
+	Perform a simple VM hibernation in Azure before VM resizng
+	VM resizing reboots the VM and fails the hibernation resume.
```
Comment (LiliDeng) [inline Testscripts/Windows/VMRESIZE-HB.ps1]: Could you please clarify more for what is expected behavior?

**Review by LiliDeng — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/VMRESIZE-HB.ps1 (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/VMRESIZE-HB.ps1]: Sorry. no. The old artifact. Fixed in bcd2282f6bd9454facee1438e9cf46ca5da42cee

File: Testscripts/Windows/VMRESIZE-HB.ps1 (hunk 2 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/VMRESIZE-HB.ps1]: Updated in bcd2282f6bd9454facee1438e9cf46ca5da42cee

File: Testscripts/Windows/VMRESIZE-HB.ps1 [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+				break
+			} else {
+				Write-LogInfo "VM is still resuming!"
+			}
+		}
+
+		if ($state){
+			Write-LogInfo "VM restart completed"
+		} else {
+			throw "VM start halt or hang, the latest state of remote connection was $state"
```
Comment (LiliDeng) [inline Testscripts/Windows/VMRESIZE-HB.ps1]: at this moment $state is $null or empty

**Review by LiliDeng — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/VMRESIZE-HB.ps1 (hunk 3 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/VMRESIZE-HB.ps1]: Fixed in 7e8639e54beeb09f5ac14bc0f1df80a6727a60b2

**Review by LiliDeng — DISMISSED** [review]

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**